engine-jdk8-tests/src/test/java/org/hibernate/validator/test/internal/engine/valuehandling/OptionalTypeAnnotationConstraintTest.java

```diff
@@ -0,0 +1,216 @@
+/*
+ * Hibernate Validator, declare and validate application constraints
+ *
+ * License: Apache License, Version 2.0
+ * See the license.txt file in the root directory or <http://www.apache.org/licenses/LICENSE-2.0>.
+ */
+package org.hibernate.validator.test.internal.engine.valuehandling;
+
+import org.hibernate.validator.constraints.CompositionType;
+import org.hibernate.validator.constraints.ConstraintComposition;
+import org.hibernate.validator.constraints.NotBlank;
+import org.hibernate.validator.testutil.constraints.NotNullTypeUse;
+import org.hibernate.validator.valuehandling.UnwrapValidatedValue;
+
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import javax.validation.*;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Null;
+import java.lang.annotation.Documented;
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+import java.util.Optional;
+import java.util.Set;
+
+import static java.lang.annotation.ElementType.*;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+import static org.hibernate.validator.testutil.ConstraintViolationAssert.*;
+import static org.hibernate.validator.testutil.ValidatorUtil.getValidator;
+
+/**
+ * @author Lukas Niemeier
```

Comment:
Since he provided the code on JIRA without any authorship I think it would be safer to remove it. In some form he still has credit via the issue itself. This avoids any problems.